main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -797,12 +839,12 @@ object Type {
     *
     * Must not be used before kinding.
     */
-  def mkOr(tpe1: Type, tpe2: Type): Type = (tpe1, tpe2) match {
+  def mkOr(tpe1: Type, tpe2: Type, loc: SourceLocation = SourceLocation.Unknown): Type = (tpe1, tpe2) match {
```

Comment:
I am not so happy that we allow a default parameters with `Unknown`.

The fact that we have `Unknown` locations is really a design mistake it Flix- they suddenly pop up in error messages and then there is total confusion. 

Are there at least some of these we could make non-default and then we would have to explicitly pass in `Unknown` -- but only in a few places? If its hundreds of places we probably have to live with the default param.

Reply:
Commit aadba6354ae91050827938a744c3f5d7c45c1e83 removes these. There were about a hundred of them at first until Intellij found more...